File: frontend_flutter/lib/modules/customers/domain/entities/customer.dart
/// Domain entity. No Flutter, no JSON.
/// Bảng Tổng hợp Danh sách Khách hàng: STT, MÃ, TÊN KHÁCH HÀNG, ĐỊA CHỈ, SỐ ĐIỆN, NGƯỜI LIÊN HỆ.
class Customer {
  const Customer({
    required this.id,
    required this.code,
    required this.name,
    required this.address,
    required this.phone,
    required this.contactPerson,
  });

  final String id;
  final String code;
  final String name;
  final String address;
  final String phone;
  final String contactPerson;
}

File: backend/internal/dto/customer.go
package dto

import "github.com/google/uuid"

// CreateCustomerRequest is the body for POST /customers.
// Code and Name are provided by Sales (NV phòng KD) when creating a new customer.
type CreateCustomerRequest struct {
	Code          string `json:"code"`
	Name          string `json:"name"`
	Address       string `json:"address"`
	Phone         string `json:"phone"`
	ContactPerson string `json:"contactPerson"`
}

// UpdateCustomerRequest is the body for PUT /customers/:id.
type UpdateCustomerRequest struct {
	Name          string `json:"name"`
	Address       string `json:"address"`
	Phone         string `json:"phone"`
	ContactPerson string `json:"contactPerson"`
}

// CustomerPayload is returned in list, create, update, get responses.
type CustomerPayload struct {
	ID            uuid.UUID `json:"id"`
	Code          string    `json:"code"`
	Name          string    `json:"name"`
	Address       string    `json:"address"`
	Phone         string    `json:"phone"`
	ContactPerson string    `json:"contactPerson"`
}
